tamr_toolbox/dataset/manage.py

```diff
@@ -0,0 +1,234 @@
+            attribute_names=attributes, attribute_types=attribute_types
+        )
+
+        # Get current dataset attributes
+        target_dataset_attributes = dataset.attributes
+        target_attribute_dict = {}
+        for attr in target_dataset_attributes.stream():
+            target_attribute_dict[attr.name] = attr
+
+        # Update attributes in dataset
+        for attribute in attribute_specs:
+            attr_spec_dict = attribute.to_dict()
+            attribute_name = attr_spec_dict["name"]
+            if attribute_name in primary_keys:
+                continue
+            elif attribute_name in target_attribute_dict.keys():
+                # This attribute already exists, update to new type
+                new_type = AttributeType(attribute.to_dict()["type"])
+                target_attribute = target_attribute_dict[attribute_name]
+                new_attr_spec = target_attribute.spec().with_type(new_type.spec())
+
+                # Remove and add attribute with new type
+                target_dataset_attributes.delete_by_resource_id(target_attribute.resource_id)
+                target_dataset_attributes.create(new_attr_spec.to_dict())
+            else:
+                # This attribute does not already exist, create
+                target_dataset_attributes.create(attr_spec_dict)
+
+        # Remove any attributes from dataset that aren't in the new list of attributes
+        attribute_names = [attr.to_dict()["name"] for attr in attribute_specs]
+        for existing_attribute in target_attribute_dict.keys():
+            if (
+                existing_attribute not in attribute_names
+                and existing_attribute not in primary_keys
+            ):
+                target_dataset_attributes.delete_by_resource_id(
+                    target_attribute_dict[existing_attribute].resource_id
+                )
+
+    return dataset
+
+
+def exists(*, client: Client, dataset: str) -> bool:
+    """Check if the dataset exists on target instance
+
+    Args:
+        target_instance: Tamr python client object for the target instance
+        dataset: The dataset name
+
+    Return:
+        True or False for if the dataset exists in target instance
+    """
+
+    try:
+        client.datasets.by_name(dataset)
+    except KeyError:
+        return False
+
+    return True
+
+
+def _create_specs(
+    *, attribute_names: List[str], attribute_types: Union[JsonDict, None]
+) -> List[AttributeSpec]:
+    """Create list of attributeSpec. Use default type if none is given
+
+    Args:
+        attribute_names: List of names of attributes
+        attribute_types: dict of attribute types, attribute name is key and type is value
+
+    Return:
+        List of AttributeSpecs
+    """
+    default_type = {
+        "baseType": "ARRAY",
+        "innerType": {"baseType": "STRING", "attributes": []},
+        "attributes": [],
+    }
+    attribute_specs = []
+    for idx in range(len(attribute_names)):
```

Comment:
should catch something if attribute_names has no length or is of type Nonetype.

Reply:
Added a check before the loop. Will create a dataset with no attributes (besides primary keys)